src/test/java/org/marketdesignresearch/mechlib/mechanism/auctions/cca/CCATest.java

```diff
@@ -258,11 +261,11 @@ public void testStepByStepCCA() {
         // Second round
         for (Bidder bidder : domain.getBidders()) {
             Bundle bestBundle = bidder.getBestBundle(cca.getCurrentPrices());
-            BundleBid bundleBid = new BundleBid(cca.getCurrentPrices().getPrice(bestBundle).getAmount(), bestBundle, UUID.randomUUID().toString());
-            Bid bid = new Bid(Sets.newHashSet(bundleBid));
-            checkBidEquality(bid, cca.proposeBid(bidder));
+            BundleValuePair bundleBid = new BundleValuePair(cca.getCurrentPrices().getPrice(bestBundle).getAmount(), bestBundle, UUID.randomUUID().toString());
+            BundleValueBid<BundleValuePair> bid = new BundleValueBid<>(Sets.newHashSet(bundleBid));
+            assertEquals(bestBundle, ((DemandBid)cca.getCurrentInteraction(bidder).proposeBid()).getDemandedBundle());
             // Submit bids one by one
-            cca.submitBid(bidder, bid);
+            cca.getCurrentInteraction(bidder).submitBid(new DemandBid(bestBundle));
```

Comment:
Another example where typecasts are needed with the new interface. As an auction can have multiple interaction types it is possible to use generics to resolve this issue.

Reply:
looks different now so resolve this conversation